packages/router/router.ts

```diff
@@ -1242,6 +1305,13 @@ export function createRouter(init: RouterInit): Router {
     let result: DataResult;
     let actionMatch = getTargetMatch(matches, location);
 
+    if (actionMatch.route.lazy) {
+      await loadLazyRouteModules([actionMatch], hasErrorBoundary, manifest);
+      if (request.signal.aborted) {
+        return { shortCircuited: true };
```

Comment:
@mjackson @ryanflorence - @markdalgleish brought up a good question here.  Normal (non-lazy) actions _cannot_ be aborted by us so once they click submit the action runs, period.  Now that we've introduced a lazy step before the action - we have the ability to cancel after click, before `action`.  Should we?  Or should we keep the consistency of "the action is always called on submission" and leave it up to user to handle the `AbortSignal` in their action?  It could feel weird to need to check `signal.aborted` as the very first line in your action though?

Reply:
I'd say the addition of the lazy API should not change how we currently call and handle actions. Let's keep the consistency and always call the action on submission.